modules/querying/pages/select-statement/index.adoc

```diff
@@ -0,0 +1,2866 @@
+--
+====
+
+==== Path edge pattern
+The path edge pattern represents the relationship between a source vertex set to a step vertex set or from a step vertex set to the next step vertex set.
+
+.EBNF for path edge pattern
+----
+pathEdgePattern := atomicEdgePattern
+                 | "(" pathEdgePattern ")"
+                 | pathEdgePattern "." pathEdgePattern
+                 | disjPattern
+                 | starPattern
+
+atomicEdgePattern  := atomicEdgeType
+        	        | atomicEdgeType ">"
+        	        | "<" atomicEdgeType
+
+atomicEdgeType := "_" | ANY | edgeSetType
+
+disjPattern := atomicEdgePattern ("|" atomicEdgePattern)*
+
+starPattern := ([atomicEdgePattern] | "(" disjPattern ")") "*" [starBounds]
+
+starBounds := CONST_INT ".." CONST_INT
+            | CONST_INT ".."
+            | ".." CONST_INT
+            | CONST_INT
+----
+
+A path edge pattern can represent one hop or repeated hops.
+A path edge pattern is denoted by `-()-`, where the relationship between vertex sets is specified between the parentheses.
+
+[discrete]
+==== Atomic edge pattern
+
+The most basic form for a path edge pattern is an atomic edge pattern.
+An atomic edge pattern can be one of the following:
+
+* `_` or `ANY`
+* An edge type, a string parameter, or a global `SetAccum` accumulator.
+
+Moreover, an atomic edge pattern can have either a left pointer `<` on the left or a right pointer `>` on the right.
+These indicate edge direction, of course. If no pointer is used, then the edge is undirected. Suppose we have 3 edge types or parameters called A, B, C.
+
+* `A>` is a rightward facing A edge
+* `<B` is a leftward facing B edge
+* C is an undirected C edge.
+If C is actually a directed edge type, then there is no match
+
+[discrete]
+==== Disjunction pattern
+Pattern disjunction allows a path edge pattern to indicate a `OR` relationship between two or more atomic patterns.
+If an edge matches any of the atomic patterns, the edge matches the path edge pattern.
+
+.EBNF for disjunction pattern
+[,ebnf]
+----
+disjPattern := atomicEdgePattern ("|" atomicEdgePattern)*
+----
+
+
+[discrete]
+==== Pattern repetition
+Star pattern is used to how the Kleene star``*`` and `min..max` range specifiers can be used to say "repeat this edge pattern from min to max times."
+See xref:tutorials:pattern-matching/repeating-a-pattern.adoc[] for a tutorial on how to use pattern repetition in a path edge pattern.
+
+.EBNF for star pattern
+[,ebnf]
+----
+starPattern := ([atomicEdgePattern] | "(" disjPattern ")") "*" [starBounds]
+
+starBounds := CONST_INT ".." CONST_INT
+            | CONST_INT ".."
+            | ".." CONST_INT
+            | CONST_INT
+----
+
+[discrete]
+==== Pattern concatenation
+The dot operator``.`` means concatenate the two edge patterns into one.
+Naturally, there must be a vertex joining the two edges, but it is omitted from the syntax.
+The dot operator is a shorthand, when you don’t care about the type of that intermediate vertex.
```

Comment:
There will be no default. The matches will include every type of vertex, and you won't be able to access those vertices since you didn't declare an alias.